### PR TITLE
Development

### PR DESCRIPTION
**Pull request by PolinaStamenova**
### Send and receive data from an API

**JavaScript:**

-  Create getPoke() function to retrieve the data from Pokemon API.
-  Create loadPoke and singlePoke functions to get more informations about the pokemons.
-  Create createElments function to display all the  pokemons with their description and like counters. With like button.
-  Create get request to get data for likes and comments.
-  Create popup that appears on comment button, including information for the pokemon and option to comment. Comment counter included.

**CSS:** 

- Add styling to the project

**HTML:**

- Add header.
-  Add footer.
- Add root div.

**Webpack:**
- Install webpack.

**Axios:**

- Use axios library to fetch data from API.

**JEST:**

-  Create test for the comments counter..

**Linters:**

- Add webhint.
- Add stylint.
- Add lighthouse.
- Add ESLint.

_Thank you for taking the time to review my PR!_ 🌵 😄

### CODE REVIEW

File: src/createElements.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const createElements = async (name, image, pokeurl) => {
+  const ul = document.getElementById('cards');
+  const li = document.createElement('li');
+  const div = document.createElement('div');
+  div.classList.add('div');
+  const like = document.createElement('i');
+  const likeCounter = document.createElement('p');
+  const commentBtn = document.createElement('button');
+  commentBtn.addEventListener('click', () => {
+    displayPopup(pokeurl);
```
Comment (hemant-soni-vst-au4) [inline src/createElements.js:16]: - [ ] Let's try to add the `hover` effect on the `comment button`. When the user hovers the comment button cursor should change to the pointer. Kindly fix this issue.

File: src/createElements.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  });
+  commentBtn.textContent = 'Comments';
+  commentBtn.classList.add('comment-btn');
+  likeCounter.textContent = await getLikes(name, likeUrl);
+  likeCounter.classList.add('likes');
+  likeCounter.id = name;
+
+  like.classList.add('far', 'fa-heart');
```
Comment (hemant-soni-vst-au4) [inline src/createElements.js:24]: - [ ] Let's try to add the `hover` effect on the `like icon`. When the user hovers the like icon cursor should change to the pointer. Kindly fix this issue.

File: src/addComment.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  const commentInput = document.createElement('input');
+  commentInput.classList.add('comment-input');
+  nameInput.type = 'text';
+  commentInput.type = 'text';
+  nameInput.placeholder = 'Your name';
+  commentInput.placeholder = 'Your comment';
+  const btn = document.createElement('button');
+  btn.innerHTML = 'Comment';
+  btn.classList.add('popup-cemment-btn');
+  btn.addEventListener('click', async () => {
```
Comment (hemant-soni-vst-au4) [inline src/addComment.js:23]: - [ ] Optional
After adding the comment, the input fields are not empty. We can improve this by adding input fields as empty on the comment button event listener. Let's try to fix it.

**Review by hemantso — CHANGES_REQUESTED** [review]
Hi @PolinaStamenova and @TSHEPO-CLOUD, 

## Highlights
- Video presentation is really good. you nicely presented the feature of the project. 👍 
- Great work on meeting most of the requirements. 💯 
- Testing of counter function.

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there! 

Alt Text

## Required Changes ♻️
_Check the comments under the review._

- [ ] As per requirements all the members should appear in the video and talk about the project. You guys can use the zoom to meet the requirements.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- It will be great if we provide a `live link` of the project in our `README` file.

Cheers,
Hemant Soni
Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PolinaStamenova** [comment]
https://drive.google.com/file/d/1hVAiHX4RsG-FlIej06DMjNAJBwqLNQ5Y/view?usp=sharing

**Review by bellom — APPROVED** [review]
## Hi @PolinaStamenova @TSHEPO-CLOUD  👏  

**Great work making all the necessary corrections from the previous review.**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

### [Optional]
The reason why we do write tests is for us to be able to debug quickly and easily, that is; once anything goes wrong in our actual code, our test should fail and point us to where we have the bug. 

_For example:_ we have a function called `counter`, we wrote a test for the counter function, test case passes, cool!
If we should introduce a bug in the `counter` function i.e make the function return `false` instead of `true`, we should expect our test to fail.

_In your case:_
we are supposed to have a `counter function`  in the codebase then we should write a jest test to test the `counter function` once the counter function is broken, the test should fail.

So you might want to work on the mocking, let's re-check [jest easy mocking](https://jestjs.io/)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PolinaStamenova** [comment]
@bellom Oluwaseun,

thank you for the review and the provided link source. 
We will check it now!

**Pull request merged**